adapter/repository/in-memory-simple/src/test/java/org/mobilitydata/gtfsvalidator/db/NoticeMemoryConsumptionTest.java

```diff
@@ -0,0 +1,140 @@
+package org.mobilitydata.gtfsvalidator.db;
+
+import org.junit.jupiter.api.Test;
+import org.mobilitydata.gtfsvalidator.usecase.notice.IntegerFieldValueOutOfRangeNotice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.base.ErrorNotice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.base.InfoNotice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.base.Notice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.base.WarningNotice;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class NoticeMemoryConsumptionTest {
+
+    private void generateNotices(MockValidationResultRepo mockValidationResultRepo, int numberOfNotices) {
+        for (int i = 0; i < numberOfNotices; i++) {
+            mockValidationResultRepo.addNotice(new IntegerFieldValueOutOfRangeNotice("filename",
+                    "fieldname",
+                    "entity_id",
+                    0,
+                    100,
+                    101)
+            );
+        }
+    }
+
+    @Test
+    public void creationOf100NoticeShouldNotRaiseException() {
+
+        MockValidationResultRepo mockValidationResultRepo = new MockValidationResultRepo();
+
+        generateNotices(mockValidationResultRepo, 100);
+        mockValidationResultRepo.getAll();
+    }
+
+    @Test
+    public void creationOf1000NoticeShouldNotRaiseException() {
+
+        MockValidationResultRepo mockValidationResultRepo = new MockValidationResultRepo();
+
+        generateNotices(mockValidationResultRepo, 1_000);
+        mockValidationResultRepo.getAll();
+
+    }
+
+    @Test
+    public void creationOf10000NoticeShouldNotRaiseException() {
+
+        MockValidationResultRepo mockValidationResultRepo = new MockValidationResultRepo();
+
+        generateNotices(mockValidationResultRepo, 10_000);
+        mockValidationResultRepo.getAll();
+
+    }
+
+    @Test
+    public void creationOf100000NoticeShouldNotRaiseException() {
+
+        MockValidationResultRepo mockValidationResultRepo = new MockValidationResultRepo();
+
+        generateNotices(mockValidationResultRepo, 100_000);
+        mockValidationResultRepo.getAll();
+
+    }
+
+    @Test
+    public void creationOf1_000_000NoticeShouldNotRaiseException() {
+
+        MockValidationResultRepo mockValidationResultRepo = new MockValidationResultRepo();
+
+        generateNotices(mockValidationResultRepo, 1_000_000);
+        mockValidationResultRepo.getAll();
+    }
+
+    @Test
+    public void creationOf3_000_000NoticeShouldRaiseException() {
+
+        MockValidationResultRepo mockValidationResultRepo = new MockValidationResultRepo();
+
+        try {
+            for (int i = 0; i < 3_000_000; i++) {
```

Comment:
Even tho I created a private method `generateNotices` to avoid code duplication. The tests did not seem to run when calling this method in this `creationOf3_000_000NoticeShouldRaiseException`. So I duplicated the code here...could be improved.

Reply:
I'd suggest trying to use the private method again after getting rid of the try block and annotating the test to expect the exception. You may need to make the private method `throws Exception {`.

Reply:
Okay, thanks. Will modify, however I suggest the private method to throw Error not an Exception since OutOfMemoryError is an Error.

Reply:
@lionel-nj Sounds good!